contract/src/Utils.sol

```diff
@@ -157,14 +175,93 @@ library Utils {
             return false;
         }
 
-        address eip712Signer = Utils.recoverStateEIP712Signer(STATE_TYPEHASH, channelId, domainSeparator, state, sig);
+        address eip712Signer = Utils.recoverStateEIP712Signer(domainSeparator, STATE_TYPEHASH, channelId, state, sig);
         if (eip712Signer == signer) {
             return true;
         }
 
         return false;
     }
 
+    /**
+     * @notice Checks if a signature is valid by calling the expected signer contract according to the ERC-1271 standard
+     * @param msgHash The hash of the message to verify the signature against
+     * @param sig The signature to verify
+     * @param expectedSigner The address of the expected signer
+     * @return True if the signature is valid, false otherwise or if signer is not a contract
+     */
+    function isValidERC1271Signature(bytes32 msgHash, bytes memory sig, address expectedSigner)
+        internal
+        view
+        returns (bool)
+    {
+        return IERC1271(expectedSigner).isValidSignature(msgHash, sig) == ERC1271_SUCCESS;
+    }
+
+    /**
+     * @notice Checks if a signature is valid by deploying the expected signer contract according to the ERC-6492 standard
```

Comment:
"is valid by deploying the expected signer" so we expect to deploy if needed?

Reply:
Correct. Let me clarify the comment a little bit to improve understanding.

Reply:
6533c781caa31f68e80774ac50f36b30f9efdaa7